plugins/inputs/mesos/mesos.go

```diff
@@ -95,6 +95,14 @@ func (m *Mesos) SetDefaults() {
 		log.Println("I! [mesos] Missing timeout value, setting default value (100ms)")
 		m.Timeout = 100
 	}
+	if client.Transport == nil {
```

Comment:
After talking to Connor, it seems like the `Transport` object attribute requires configuration to be read (and an instance of mesos object to exist). This implies that we have to set it after the `Init` on the mesos object